applicationset/generators/scm_provider.go

```diff
@@ -58,6 +60,21 @@ func (g *SCMProviderGenerator) GetTemplate(appSetGenerator *argoprojiov1alpha1.A
 	return &appSetGenerator.SCMProvider.Template
 }
 
+// Returns an error in case url is not the default provider URL or allowed with allowedScmProviders. If allowedScmProviders is empty it means all are allowed
+func ScmProviderAllowed(url string, allowedScmProviders []string) error {
```

Comment:
Maybe lowercase? The smaller the public API the better imo.

Reply:
I am using it in the pull request generator. Do you want me to extract this in utility class ?

Reply:
Ah! Eh, your call. Whatever seems tidiest. :-)